### ISSUE TITLE
FeatureFlag feature

### ISSUE THREAD

**Issue by lxgr-linux** [DecentralCardGame/Cardchain#153]
I'm playing with the idea of adding a FeatureFlag service that enables us to switch on features via a governance vote on the chain just in time, when infrastructure in frontend/gameclient is build.

Concrete usecase:
Council overhaul can be build and merged into master and deployed.
Can be tested on testnet while frontend/gameclient are developed to have the gameclient boys not wait on cardchain pull requests.

When council is finished and working in testnet, council can just be switched on via governance.
This would reduce riscs of big PRs breaking production and reduces bigger PRs laying around for months.

**Comment by lxgr-linux** [comment]
@patrickwieth, thoughts?

**Comment by patrickwieth** [comment]
The idea is good, one remark:

As far as I understand it, the idea is to have an abstraction that allows to switch off a feature that is build under this abstraction or on if desired. Now for the concrete example this means we can switch off or on the council as necessary. However in reality we do not really want to switch it on/off but rather switch between 2 different methods, which are council or instant pass. If the abstraction just deactivates a feature, then I would expect it just isn't used anymore and not replaced by something that does another action instead (which is necessary here to make sense). 

So for the concrete example maybe just a parameter that is called skipCouncil might be sufficient. However if we find more uses for that FeatureFlag service, then it might make sense?

**Comment by lxgr-linux** [comment]
> The idea is good, one remark:
> 
> As far as I understand it, the idea is to have an abstraction that allows to switch off a feature that is build under this abstraction or on if desired. Now for the concrete example this means we can switch off or on the council as necessary. However in reality we do not really want to switch it on/off but rather switch between 2 different methods, which are council or instant pass. If the abstraction just deactivates a feature, then I would expect it just isn't used anymore and not replaced by something that does another action instead (which is necessary here to make sense).
> 
> So for the concrete example maybe just a parameter that is called skipCouncil might be sufficient. However if we find more uses for that FeatureFlag service, then it might make sense?

That's exactly how it was meant. No council would mean instant pass.

**Comment by patrickwieth** [comment]
Good. Idea good. Other ideas, emotional daaamage.
